src/itype.h

```diff
@@ -395,51 +394,51 @@ struct itype {
     // name and name_plural are not translated automatically
     // nname() is used for display purposes
     std::string name;        // Proper name, singular form, in American English.
     std::string name_plural; // name, plural form, in American English.
 public:
+    std::string snippet_category;
     std::string description; // Flavor text
 
-    char sym;       // Symbol on the map
-    nc_color color; // Color on the map (color.h)
+    std::map<std::string, int> qualities; //Tool quality indicators
+    std::map<std::string, std::string> properties;
 
     // What we're made of (material names). .size() == made of nothing.
     // MATERIALS WORK IN PROGRESS.
     std::vector<std::string> materials;
+    std::vector<use_function> use_methods; // Special effects of use
 
-    phase_id phase; //e.g. solid, liquid, gas
-
-    unsigned int volume; // Space taken up by this item
-    int stack_size;      // How many things make up the above-defined volume (eg. 100 aspirin = 1 volume)
-    unsigned int weight; // Weight in grams. Assumes positive weight. No helium, guys!
-    std::map<std::string, int> qualities; //Tool quality indicators
-    std::map<std::string, std::string> properties;
-
+    std::set<std::string> item_tags;
+    std::set<std::string> techniques;
+    
     // Explosion that happens when the item is set on fire
     explosion_data explosion_on_fire_data;
-    bool explode_in_fire() const
-    {
-        return explosion_on_fire_data.power >= 0;
-    }
 
-    signed int melee_dam; // Bonus for melee damage; may be a penalty
-    signed int melee_cut; // Cutting damage in melee
-    signed int m_to_hit;  // To-hit bonus for melee combat; -5 to 5 is reasonable
+    phase_id phase      = SOLID; // e.g. solid, liquid, gas
+    unsigned price      = 0; // Its value
+    unsigned volume     = 0; // Space taken up by this item
+    int      stack_size = 0; // How many things make up the above-defined volume (eg. 100 aspirin = 1 volume)
+    unsigned weight     = 0; // Weight in grams. Assumes positive weight. No helium, guys!
 
-    std::set<std::string> item_tags;
-    std::set<std::string> techniques;
+    int melee_dam = 0; // Bonus for melee damage; may be a penalty
+    int melee_cut = 0; // Cutting damage in melee
+    int m_to_hit  = 0;  // To-hit bonus for melee combat; -5 to 5 is reasonable
 
-    unsigned int light_emission;   // Exactly the same as item_tags LIGHT_*, this is for lightmap.
+    unsigned light_emission = 0;   // Exactly the same as item_tags LIGHT_*, this is for lightmap.
 
-    const item_category *category; // category pointer or NULL for automatic selection
+    const item_category *category = nullptr; // category pointer or NULL for automatic selection
 
-    std::string snippet_category;
+    nc_color color = c_white; // Color on the map (color.h)
+    char sym = '#';       // Symbol on the ma
+
+    bool explode_in_fire() const
+    {
+        return explosion_on_fire_data.power >= 0;
```

Comment:
Maybe it means a 0-size explosion, i.e. the item is destroyed and nothing
else happens. Seems a bit odd, but there you have it.

On Sat, Feb 21, 2015 at 2:31 PM, Davek <notifications@github.com> wrote:

> In src/itype.h
> <https://github.com/CleverRaven/Cataclysm-DDA/pull/11285#discussion_r25121869>
> :
>
> >
> > -    std::string snippet_category;
> > +    nc_color color = c_white; // Color on the map (color.h)
> > +    char sym = '#';       // Symbol on the ma
> > +
> > +    bool explode_in_fire() const
> > +    {
> > +        return explosion_on_fire_data.power >= 0;
>
> I wonder then, as a power of 0 doesn't seem like it should explode, but
> that is not within the confines of this PR, so nvm. :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CleverRaven/Cataclysm-DDA/pull/11285/files#r25121869>.
>